core/arch/arm/kernel/generic_boot.c

```diff
@@ -471,17 +471,67 @@ static void init_runtime(unsigned long pageable_part __unused)
 }
 #endif
 
-#if defined(CFG_DT) && !defined(CFG_EMBED_DTB)
-void *get_dt_blob(void)
+#if defined(CFG_DT)
+#if defined(CFG_EMBED_DTB)
+static void *get_embedded_dt_blob(void)
 {
+	static bool checked;
+
 	assert(cpu_mmu_enabled());
-	return dt_blob_addr;
+
+	if (!checked) {
+		if (fdt_check_header(embedded_secure_dtb))
+			panic("Invalid embedded DTB");
+
+		checked = true;
+	}
+
+	return embedded_secure_dtb;
+}
+#else
+static void *get_embedded_dt_blob(void)
+{
+	return NULL;
+}
+#endif /*CFG_EMBED_DTB*/
+
+void *get_sec_dtb(void)
+{
+	void *fdt = get_embedded_dt_blob();
+
+	if (fdt)
+		return fdt;
+
+	if (tee_vbuf_is_sec(external_dt_blob, CFG_DTB_MAX_SIZE))
```

Comment:
Here is a bit late to check if the DTB is to be considered secure or not. The classification should be done when the pointer is saved.

Reply:
Looks like we don't need `get{sec|ns}_dtb()` but rather `get_{embedded|external}_dtb()` adn still `get_dtb()` with same behaviour (emb then ext).